### DISCUSSION TITLE
clock calibration and software updates

### DISCUSSION THREAD

**Discussion by Chomemel** [sanni/cartreader#680] [Need help]
My cart reader runs with v9.1. To read snes SA-1 games I had to calibrate the clock generator.

If I update the sofware, will I have to calibrate the clock generator again? Or wil it keep the calibration I did?

**Comment by sanni** [comment]
In Cart_Reader.ino change this line like so:
````
// Use calibration data from snes_clk.txt
#define clockgen_calibration
````

Then the calibration will be read out of the snes_clk.txt that should be on your SD card from the last time you calibrated the clock generator.
You can check if you have that file and note your current calibration down too, just in case.

**Comment by Chomemel** [comment]
Ok, thanks for th tip, Sanni.
I'll try it when I update the software.
